CMake/External_OpenCV.cmake

```diff
@@ -173,16 +208,17 @@ else()
   list(APPEND OpenCV_EXTRA_BUILD_FLAGS -DBUILD_JPEG=ON)
 endif()
 
-if (OpenCV_SELECT_VERSION VERSION_EQUAL 2.4.11)
+
+# If a patch file exists for this version, apply it
+set (OpenCV_patch ${fletch_SOURCE_DIR}/Patches/OpenCV/${OpenCV_SELECT_VERSION})
+if (EXISTS ${OpenCV_patch})
   set(OPENCV_PATCH_COMMAND ${CMAKE_COMMAND}
-    -DOpenCV_patch:PATH=${fletch_SOURCE_DIR}/Patches/OpenCV2
+    -DOpenCV_patch:PATH=${OpenCV_patch}
     -DOpenCV_source:PATH=${fletch_BUILD_PREFIX}/src/OpenCV
-    -P ${fletch_SOURCE_DIR}/Patches/OpenCV2/Patch.cmake)
+    -P ${OpenCV_patch}/Patch.cmake
+    )
 else()
-  set(OPENCV_PATCH_COMMAND ${CMAKE_COMMAND}
-    -DOpenCV_patch:PATH=${fletch_SOURCE_DIR}/Patches/OpenCV3
-    -DOpenCV_source:PATH=${fletch_BUILD_PREFIX}/src/OpenCV
-    -P ${fletch_SOURCE_DIR}/Patches/OpenCV3/Patch.cmake)
+  set(OPENCV_PATCH_COMMAND "")
```

Comment:
The reason its there is because my `CMake` skills are intermediate and I didn't know if an undefined variable threw an error or evaluated to an empty string. I tend to err on the side of including an `else` cases when I code, so this is a result of that and my laziness for not checking the thing I wasn't sure about.

However, now that you mention the second point, it probably should be there to clobber the value of `OPENCV_PATCH_COMMAND` if the versions switch. So, I'll amend my previous statement and say I wrote that code for very conscious and good reasons (I'll add a comment explaining the behavior).